lib/auth/result.rb

```diff
@@ -165,7 +156,27 @@ def to_client_hash
 
   private
 
+  def staged_user
+    @staged_user ||= begin
+      if email.present? && email_valid
+        User.where(staged: true).find_by_email(email)
+      else
+        nil
+      end
+    end
```

Comment:
Is it supposed to memoize only non-nil values? If not:

```suggestion
    return @staged_user if defined?(@staged_user)
    if email.present? && email_valid
      @staged_user = User.where(staged: true).find_by_email(email)
    end
```